### PR TITLE
[Tests-Only] Added test for account update

### PR DESCRIPTION
**Pull request by kiranparajuli589**
Added unit test for update account.
**Related:** Fixes https://github.com/owncloud/ocis-accounts/issues/29

### CODE REVIEW

**Comment by IljaN** [comment]
Also added some here: https://github.com/owncloud/ocis-accounts/pull/69/files#diff-fd403341fdde4ca4d1dd561f76508ae5

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 1]
```diff
@@ -244,6 +244,9 @@ func assertAccountsSame(t *testing.T, acc1, acc2 *proto.Account) {
 	assert.Equal(t, acc1.DisplayName, acc2.DisplayName)
 	assert.Equal(t, acc1.PreferredName, acc2.PreferredName)
 	assert.Equal(t, acc1.UidNumber, acc2.UidNumber)
+	assert.Equal(t, acc1.OnPremisesSamAccountName, acc2.OnPremisesSamAccountName)
+	assert.Equal(t, acc1.GidNumber, acc2.GidNumber)
+	//assert.Equal(t, acc1.MemberOf, acc2.MemberOf)
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:249]: why it this commented out?

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 2]
```diff
@@ -423,6 +439,89 @@ func TestCreateAccountInvalidUserName(t *testing.T) {
...
+	tests := []struct {
+		name        string
+		userAccount *proto.Account
+		updateMask  []string
+		expectedAccount    *proto.Account
+	}{
+		{
+			"Update user",
+			&proto.Account{
+				Id:            "f9149a32-2b8e-4f04-9e8d-937d81712b9a", // id of user1
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:454]: seems to be the same for every test, no need to repeat it

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 3]
```diff
@@ -423,6 +439,89 @@ func TestCreateAccountInvalidUserName(t *testing.T) {
...
+				DisplayName:   "Hello Kiran",
+				PreferredName: "Kiran Hello",
+			},
+			[]string{
+				"DisplayName",
+				"PreferredName",
+			},
```
Comment (individual-it) [inline pkg/proto/v0/accounts.pb.micro_test.go:461]: seems to be the same for every test, no need to repeat it

**Comment by kiranparajuli589** [comment]
requested changes have been made @individual-it, please review again!

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 4]
```diff
@@ -423,6 +438,129 @@ func TestCreateAccountInvalidUserName(t *testing.T) {
...
+			if err == nil {
+				t.Fatalf("Expected error while updating non updatable field, but found none.")
+			}
+			assert.IsType(t, &proto.Account{}, res)
+			assert.Empty(t, res)
+			assert.Error(t, err)
+
+			var e *merrors.Error
+
+			if errors.As(err, &e) {
```
Comment (dpakach) [inline pkg/proto/v0/accounts.pb.micro_test.go:554]: I think you can also fail the test if the error doesn't matches

File: pkg/proto/v0/accounts.pb.micro_test.go [hunk 5]
```diff
@@ -423,6 +438,129 @@ func TestCreateAccountInvalidUserName(t *testing.T) {
...
+			"Update user with unicode data",
+			&proto.Account{
+				DisplayName:                 "एलिस हेन्सेन",
+				PreferredName:               "अद्भुत एलिस",
+				OnPremisesDistinguishedName: "एलिस",
+				UidNumber:                   20010,
+				GidNumber:                   30001,
+				Mail:                        "एलिस@उदाहरण.com",
+			},
+		},
```
Comment (dpakach) [inline pkg/proto/v0/accounts.pb.micro_test.go:469]: You can also try more examples with different update values such as empty string, numbers, random string. Some of those might not work.

**Pull request merged**